File: Assignment6/src/main/java/edu/miu/assignment6/entity/Post.java
package edu.miu.assignment6.entity;

import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
@Table(name="posts")
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.TABLE, generator = "post_id")
    @TableGenerator(name = "post_id", table = "post_id", initialValue = 5)
    @Column(nullable = false)
    long id;
     String title;
     String content;
     String author;
     @ManyToOne
     @JoinColumn
     User user;

     @OneToMany(cascade = CascadeType.ALL)
     @JoinColumn(name = "post_id")
     List<Comment> comments;
}

File: assignment9/src/Component/Post/Post.js


const Post = (props) => {
    return (
        <div className="Content" onClick={props.setSelected}>
            <h3>Id: {props.id}</h3>
            <h3>Title: {props.title}</h3>
            <h3>Author: {props.author}</h3>
           <h3>Content: {props.content}</h3>
        </div>
    );
}

export default Post;